connectivity/drivers/ble/FEATURE_BLE/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_MCU_NRF52840/mbed_lib.json

```diff
@@ -1,22 +1,6 @@
 {
     "name": "cordio-ll-nrf52840",
     "config": {
-        "phy-coded-support": {
```

Comment:
Wouldn't it be possible to preserve backward compatibility by keeping these parameters and overriding the cordio LL one if defined ? 

For example in the HCI driver:

```c++
#if defined(MBED_CONF_CORDIO_LL_NRF52840_MAX_ACL_SIZE)
#warning "Please override the confirguration parameter cordio-ll.max-acl-size instead of cordio-ll-nrf52840.max-acl-size"
#undef MBED_CONF_CORDIO_LL_MAX_ACL_SIZE
#define MBED_CONF_CORDIO_LL_MAX_ACL_SIZE MBED_CONF_CORDIO_LL_NRF52840_MAX_ACL_SIZE
``` 

It would speed up this PR integration.

Reply:
One of the reasons for this PR was to get rid of those confusing redefinitions that are already being used in NRFCordioHCIDriver.cpp. In our mbed_config.h we have values specific for the NRF52840 which overrule the normal cordio_ll values.

It might be better to split the PR into two separate PRs. One will add the options the configure the buffers used in NRFCordioHCIDriver.cpp which stay compatible. The other will be breaking due to the cleaner way of configuring the ACL size etc.